src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -1588,9 +1586,7 @@ void MergeTreeDataMergerMutator::mutateSomePartColumns(
     MergedColumnOnlyOutputStream out(
         new_data_part,
         mutation_header,
-        /* sync = */ false,
```

Comment:
Maybe it was an attempt for initial support of syncing data to disk?

Reply:
It also was used only for alters.